pkg/api/mlflow/service/artifact/service.go

```diff
@@ -58,3 +63,40 @@ func (s Service) ListArtifacts(
 
 	return run.ArtifactURI, artifacts, nil
 }
+
+// GetArtifact handles business logic of `GET /artifacts/get` endpoint.
+func (s Service) GetArtifact(
+	ctx context.Context, req *request.GetArtifactRequest,
+) (io.ReadCloser, error) {
+	if err := ValidateGetArtifactRequest(req); err != nil {
+		return nil, err
+	}
+
+	run, err := s.runRepository.GetByID(ctx, req.GetRunID())
+	if err != nil {
+		return nil, api.NewInternalError("unable to find run '%s': %s", req.GetRunID(), err)
+	}
+	if run == nil {
+		return nil, api.NewResourceDoesNotExistError("unable to find run '%s'", req.GetRunID())
+	}
+	artifactStorage, err := s.artifactStorageFactory.GetStorage(run.ArtifactURI)
+	if err != nil {
+		return nil, api.NewInternalError("run with id '%s' has unsupported artifact storage", run.ID)
+	}
+
+	artifactReader, err := artifactStorage.Get(
+		run.ArtifactURI, req.Path,
+	)
+	if err != nil {
+		msg := fmt.Sprintf(
+			"error getting artifact object for URI: %s",
+			filepath.Join(run.ArtifactURI, req.Path),
+		)
+		if errors.Is(err, fs.ErrNotExist) {
+			return nil, api.NewResourceDoesNotExistError(msg)
+		} else {
```

Comment:
you probably don't need this `else` statement. just return `api.NewInternalError`. also it will be still nice to track original error here, since it has a lot of information about error, so please process it.

Reply:
@jgiannuzzi and I discussed this, it probably needs a refactoring of the API errors to wrap an error, which is beyond the scope of this PR.

Reply:
ok, what about `else`?